muikku-persistence/src/main/java/fi/otavanopisto/muikku/model/users/UserSchoolDataIdentifier.java

```diff
@@ -86,10 +84,12 @@ public boolean hasRole(EnvironmentRoleArchetype role) {
   
   @Transient
   public boolean hasAnyRole(EnvironmentRoleArchetype ... roles) {
-    if (getRole() != null) {
-      EnvironmentRoleArchetype userRole = getRole().getArchetype();
-      
-      return userRole != null && ArrayUtils.contains(roles, userRole);
+    if (getRoles() != null) {
+      for (EnvironmentRoleEntity roleEntity : getRoles()) {
```

Comment:
Would `Collections.disjoint(roles, getRoles())` apply here?

Reply:
Wasn't as straightforward as `roles` is not a collection and `getRoles()` is a collection of entities, did a stream approach instead. (This should be thoroughly tested tho.)